### PR TITLE
Feature/function promote

### PR DESCRIPTION
**Pull request by FransVanEk**
implemented function

### CODE REVIEW

File: JLio.Commands/Set.cs [hunk 1]
```diff
@@ -45,10 +45,17 @@ public override JLioExecutionResult Execute(JToken dataContext, IExecutionContex
                 return new JLioExecutionResult(false, dataContext);
             }
 
-            var targetPath = JsonPathMethods.SplitPath(Path);
-            SetValueToObjectItems(dataContext, targetPath);
-            executionContext.LogInfo(CoreConstants.CommandExecution,
-                $"{CommandName}: completed for {targetPath.Elements.ToPathString()}");
+            //var targetPath = JsonPathMethods.SplitPath(Path);
```
Comment (cristinamudura) [inline JLio.Commands/Set.cs:48]: please remove the commented line

File: JLio.Core/Contracts/IFunction.cs [hunk 2]
```diff
@@ -9,8 +9,47 @@ public interface IFunction
...
+        public SelectedTokens Data { get; }
+
+        public bool Success { get; }
+
+        public static JLioFunctionResult Failed(SelectedTokens data)
+        {
+            return new JLioFunctionResult(false, data);
+        }
+
+        public static JLioFunctionResult SuccessFull(SelectedTokens data)
```
Comment (cristinamudura) [inline JLio.Core/Contracts/IFunction.cs:40]: please rename to Successful

File: JLio.Core/Contracts/IFunction.cs [hunk 3]
```diff
@@ -9,8 +9,47 @@ public interface IFunction
...
+        {
+            return new JLioFunctionResult(true, data);
+        }
+
+        public static JLioFunctionResult Failed(JToken data)
+        {
+            return new JLioFunctionResult(false, new SelectedTokens(data));
+        }
+
+        public static JLioFunctionResult SuccessFull(JToken data)
```
Comment (cristinamudura) [inline JLio.Core/Contracts/IFunction.cs:50]: please rename to successful

File: JLio.Core/Extensions/JsonPathItemsFetcher.cs [hunk 4]
```diff
@@ -7,6 +7,7 @@ namespace JLio.Core.Extensions
     public class JsonPathItemsFetcher : IItemsFetcher
```
Comment (cristinamudura) [inline JLio.Core/Extensions/JsonPathItemsFetcher.cs:8]: remove space

File: JLio.Core/Models/FunctionSupportedValue.cs [hunk 5]
```diff
@@ -12,12 +12,12 @@ public FunctionSupportedValue(IFunction function)
...
 
-        public JToken GetValue(JToken currentToken, JToken dataContext, IExecutionContext context)
+        public SelectedTokens GetValue(JToken currentToken, JToken dataContext, IExecutionContext context)
         {
-            var result = Function.Execute(currentToken, dataContext, options);
-            if (result.Data == null) return JValue.CreateNull();
-            if (result.Data.Type == JTokenType.String)
-                return result.Data.ToString().Trim(CoreConstants.StringIndicator);
+            var result = Function.Execute(currentToken, dataContext, context);
+            if (result.Success == false)
```
Comment (cristinamudura) [inline JLio.Core/Models/FunctionSupportedValue.cs:18]: use !result.Success

File: JLio.Core/Models/FunctionSupportedValue.cs [hunk 6]
```diff
@@ -12,12 +12,12 @@ public FunctionSupportedValue(IFunction function)
...
+                context.LogError(CoreConstants.FunctionExecution, "Execute of function failed");
```
Comment (cristinamudura) [inline JLio.Core/Models/FunctionSupportedValue.cs:19]: Do we know which function it is? so we can add it in the log?

File: JLio.Core/Models/JLioExecutionResult.cs [hunk 7]
```diff
@@ -13,5 +13,15 @@ public JLioExecutionResult(bool success, JToken data)
         public JToken Data { get; }
 
         public bool Success { get; }
+
+        public static JLioExecutionResult Failed(JToken data)
+        {
+            return new JLioExecutionResult(false, data);
+        }
+
+        public static JLioExecutionResult SuccessFull(JToken data)
```
Comment (cristinamudura) [inline JLio.Core/Models/JLioExecutionResult.cs:22]: please rename to Successful

File: JLio.Core/Models/FunctionSupportedValue.cs (hunk 6 above)
Comment (FransVanEk) [reply JLio.Core/Models/FunctionSupportedValue.cs:19]: this is fixed

**Pull request merged**